# Getting Started
Example scripts for common analyses. We're starting this directory to keep track of example scripts we've made for Goldberg Lab undergraduate researchers, but please feel free to contribute anything that might be useful to future lab members.

## Public datasets:
- 1000 Genomes resequencing data can be accessed here: https://www.internationalgenome.org/data-portal/data-collection/30x-grch38 

## Software:
- plink v1.9 https://www.cog-genomics.org/plink/1.9/
- ADMIXTURE v1.3.0 https://dalexander.github.io/admixture/download.html
- R v3.6.1 https://www.r-project.org/

## Summaries/walkthroughs within this directory:
- [Prep1000G_MalariaProject_summary.md](./Prep1000G_MalariaProject_summary.md) written Oct 2020 to assist with looking at possible selection at malaria-interacting loci in 1000G admixed populations. contains some guidance basic PLINK commands and PCA/ADMIXTURE.
- [ADMIXTURE_allele-freq_combined.pdf](./ADMIXTURE_allele-freq_combined.pdf) written Oct 2020. Example walk-through for plotting ADMIXTURE results in R and comparing observed to expected allele frequencies.
- [Intro to Unix_Cluster Computing.pdf](./Intro%20to%20Unix_Cluster%20Computing.pdf) written Aug 2020. Notes for introduction to Unix/Terminal and the Duke Compute Cluster architecture.
- [SLiM_jobarray_example.pdf](./SLiM_jobarray_example.pdf) written March 2021. Notes on how to use job arrays to run and keep track of many SLiM simulations on the cluster.
